From 7eacdaea71ae02c74875816576b707877545aa4b Mon Sep 17 00:00:00 2001
From: thampan <joby.thampan@intel.com>
Date: Mon, 11 Jan 2021 15:40:21 +0800
Subject: [PATCH] DRV_DPM_SW-429: ExtVLAN api in DPM unified

                - Mib Api removed from DPM
                - GSWIP Api's moved from Misc to gswip.c
---
 drivers/net/datapath/dpm/gswip31/Makefile          |   4 -
 .../net/datapath/dpm/gswip31/datapath_ext_vlan.c   |  17 +-
 drivers/net/datapath/dpm/gswip31/datapath_gswip.c  | 203 ++++++++++++
 drivers/net/datapath/dpm/gswip31/datapath_mib.c    | 363 ---------------------
 drivers/net/datapath/dpm/gswip31/datapath_mib.h    |  15 -
 drivers/net/datapath/dpm/gswip31/datapath_misc.c   | 219 +------------
 drivers/net/datapath/dpm/gswip31/datapath_misc.h   |  15 +-
 .../datapath/dpm/gswip31/datapath_tc_asym_vlan.c   |  36 +-
 .../net/datapath/mib_counters/dp_mib_counters.c    |   2 +-
 9 files changed, 247 insertions(+), 627 deletions(-)

diff --git a/drivers/net/datapath/dpm/gswip31/Makefile b/drivers/net/datapath/dpm/gswip31/Makefile
index 395f7b89df96..cde8d7420852 100644
--- a/drivers/net/datapath/dpm/gswip31/Makefile
+++ b/drivers/net/datapath/dpm/gswip31/Makefile
@@ -3,10 +3,6 @@ obj-$(CONFIG_INTEL_DATAPATH) += datapath_ppv4.o
 obj-$(CONFIG_INTEL_DATAPATH) += datapath_ppv4_api.o
 obj-$(CONFIG_INTEL_DATAPATH) += datapath_rx.o datapath_tx.o
 
-ifneq ($(CONFIG_INTEL_DATAPATH_HAL_GSWIP31_MIB),)
-obj-$(CONFIG_INTEL_DATAPATH) += datapath_mib.o
-endif
-
 ifneq ($(CONFIG_INTEL_DATAPATH_CPUFREQ),)
 obj-$(CONFIG_INTEL_DATAPATH) += datapath_coc.o
 endif
diff --git a/drivers/net/datapath/dpm/gswip31/datapath_ext_vlan.c b/drivers/net/datapath/dpm/gswip31/datapath_ext_vlan.c
index 733e8f1bb02e..0a7159c96074 100644
--- a/drivers/net/datapath/dpm/gswip31/datapath_ext_vlan.c
+++ b/drivers/net/datapath/dpm/gswip31/datapath_ext_vlan.c
@@ -1,25 +1,14 @@
 // SPDX-License-Identifier: GPL-2.0
-/* Copyright (C) Intel Corporation
- * Author: Shao Guohua <guohua.shao@intel.com>
+/*
+ * Copyright (C) 2016-2020 Intel Corporation
+ * Copyright (C) 2021 MaxLinear
  */
 
-#ifdef NON_LINUX
-#include <stdlib.h>
-#include <errno.h>
-#include <string.h>
-#include "lantiq_gsw.h"
-#include "lantiq_gsw_api.h"
-#include "gsw_flow_ops.h"
-/* Adaption */
-#define kfree(x)	free(x)
-#define kmalloc(x, y)	malloc(x)
-#else
 #include <linux/kernel.h>
 #include <linux/types.h>
 #include <linux/etherdevice.h>
 #include <net/datapath_api.h>
 #include "../datapath.h"
-#endif
 
 /* Structure only used by function set_gswip_ext_vlan */
 struct priv_ext_vlan {
diff --git a/drivers/net/datapath/dpm/gswip31/datapath_gswip.c b/drivers/net/datapath/dpm/gswip31/datapath_gswip.c
index cf307d8e3558..1613abd8e87e 100644
--- a/drivers/net/datapath/dpm/gswip31/datapath_gswip.c
+++ b/drivers/net/datapath/dpm/gswip31/datapath_gswip.c
@@ -1825,3 +1825,206 @@ int dp_gsw_set_bp_attr(struct dp_bp_attr *bp_attr, int bp, uint32_t flag)
 
 	return DP_SUCCESS;
 }
+
+#define Y GSW_DROP_PRECEDENCE_YELLOW
+#define G GSW_DROP_PRECEDENCE_GREEN
+
+GSW_QoS_colorRemarkingEntry_t color_remark[] = {
+	{
+		GSW_MARKING_PCP_8P0D,
+		{0, 2, 4, 6, 8, 10, 12, 14, 1, 3, 5, 7, 9, 11, 13, 15},
+	},
+	{
+		GSW_MARKING_PCP_7P1D,
+		{0, 2, 4, 6, 10, 10, 12, 14, 1, 3, 5, 7, 9, 9, 13, 15},
+	},
+	{
+		GSW_MARKING_PCP_6P2D,
+		{0, 2, 6, 6, 10, 10, 12, 14, 1, 3, 5, 5, 9, 9, 13, 15},
+	},
+	{
+		GSW_MARKING_PCP_5P3D,
+		{2, 2, 6, 6, 10, 10, 12, 14, 1, 1, 5, 5, 9, 9, 13, 15},
+	},
+	{
+		GSW_MARKING_DSCP_AF,
+		{10, 18, 26, 34, 34, 34, 34, 34, 12, 20, 28, 36, 36, 36, 36, 36},
+	},
+};
+
+GSW_QoS_colorMarkingEntry_t color_mark[] = {
+	{
+		GSW_MARKING_PCP_5P3D,
+		{0, 0, 0, 0, 2, 2, 2, 2, 4, 4, 4, 4, 6, 6, 7, 7},
+		{Y, Y, G, Y, Y, Y, G, Y, Y, Y, G, Y, G, Y, G, Y},
+	},
+	{
+		GSW_MARKING_PCP_6P2D,
+		{0, 0, 1, 1, 2, 2, 2, 2, 4, 4, 4, 4, 6, 6, 7, 7},
+		{G, Y, G, Y, Y, Y, G, Y, Y, Y, G, Y, G, Y, G, Y},
+	},
+	{
+		GSW_MARKING_PCP_7P1D,
+		{0, 0, 1, 1, 2, 2, 3, 3, 4, 4, 4, 4, 6, 6, 7, 7},
+		{G, Y, G, Y, G, Y, G, Y, Y, Y, G, Y, G, Y, G, Y},
+	},
+	{
+		GSW_MARKING_PCP_8P0D,
+		{0, 0, 1, 1, 2, 2, 3, 3, 4, 4, 5, 5, 6, 6, 7, 7},
+		{G, Y, G, Y, G, Y, G, Y, G, Y, G, Y, G, Y, G, Y},
+	},
+	{
+		GSW_MARKING_DSCP_AF,
+		{
+			0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+			0, 0, 0, 0, 1, 0, 1, 0, 1, 0, 0, 0, 2, 0,
+			2, 0, 2, 0, 0, 0, 3, 0, 3, 0, 3, 0, 0, 0,
+			0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+			0, 0, 0, 0, 0, 0, 0, 0
+		},
+		{
+			Y, Y, Y, Y, Y, Y, Y, Y, Y, Y, G, Y, Y, Y, Y, Y,
+			Y, Y, G, Y, Y, Y, Y, Y, Y, Y, G, Y, Y, Y, Y, Y,
+			Y, Y, G, Y, Y, Y, Y, Y, Y, Y, Y, Y, Y, Y, Y, Y,
+			Y, Y, Y, Y, Y, Y, Y, Y, Y, Y, Y, Y, Y, Y, Y, Y
+		},
+	},
+};
+
+int dp_gsw_color_table_set(int inst)
+{
+	struct core_ops *gsw_ops = dp_port_prop[inst].ops[0];
+	GSW_QoS_colorRemarkingEntry_t col_remark = {0};
+	GSW_QoS_colorMarkingEntry_t col_mark = {0};
+	struct qos_ops *gsw_qos;
+	int remark_size = ARRAY_SIZE(color_remark);
+	int mark_size = ARRAY_SIZE(color_mark);
+	int i = 0;
+
+	gsw_qos = &gsw_ops->gsw_qos_ops;
+
+	for (i = 0; i < remark_size; i++) {
+		memset(&col_remark, 0, sizeof(GSW_QoS_colorRemarkingEntry_t));
+		memcpy(&col_remark, &color_remark[i],
+		       sizeof(GSW_QoS_colorRemarkingEntry_t));
+		if (gsw_qos->QOS_ColorReMarkingTableSet(gsw_ops, &col_remark)) {
+			pr_err("%s: GSW_QOS_COLOR_REMARKING_CFG_SET failed\n",
+			       __func__);
+			return DP_FAILURE;
+		}
+	}
+
+	for (i = 0; i < mark_size; i++) {
+		memset(&col_mark, 0, sizeof(GSW_QoS_colorMarkingEntry_t));
+		memcpy(&col_mark, &color_mark[i],
+		       sizeof(GSW_QoS_colorMarkingEntry_t));
+		if (gsw_qos->QOS_ColorMarkingTableSet(gsw_ops, &col_mark)) {
+			pr_err("%s: GSW_QOS_COLOR_REMARKING_CFG_SET failed\n",
+			       __func__);
+			return DP_FAILURE;
+		}
+	}
+
+	return DP_SUCCESS;
+}
+
+int dp_get_gswip_cap(struct dp_hw_cap *cap, int flag)
+{
+	struct core_ops *gsw_ops;
+	GSW_cap_t parm;
+	GSW_register_t reg;
+
+	gsw_ops = gsw_get_swcore_ops(0);
+	if (!gsw_ops) {
+		pr_err("%s swcore ops is NULL\n", __func__);
+		return DP_FAILURE;
+	}
+
+	reg.nRegAddr = GSWIP_VER_REG_OFFSET;
+	if (gsw_ops->gsw_common_ops.RegisterGet(gsw_ops, &reg))
+		return DP_FAILURE;
+
+	if (reg.nData == 0x031) {
+		cap->info.type = GSWIP31_TYPE;
+		cap->info.ver = GSWIP31_VER;
+	} else if (reg.nData == 0x032) {
+		cap->info.type = GSWIP32_TYPE;
+		cap->info.ver = GSWIP32_VER;
+	}
+
+	/* Max number of CTP */
+	parm.nCapType = GSW_CAP_TYPE_CTP;
+	gsw_ops->gsw_common_ops.CapGet(gsw_ops, &parm);
+	cap->info.cap.max_num_subif = parm.nCap;
+
+	/* Max number of Bridge Port */
+	parm.nCapType = GSW_CAP_TYPE_BRIDGE_PORT;
+	gsw_ops->gsw_common_ops.CapGet(gsw_ops, &parm);
+	cap->info.cap.max_num_bridge_port = parm.nCap;
+
+	/* Max number of Physical Ethernet Port */
+	parm.nCapType = GSW_CAP_TYPE_PORT;
+	gsw_ops->gsw_common_ops.CapGet(gsw_ops, &parm);
+	cap->info.cap.max_eth_port = parm.nCap;
+
+	/* Max number of Virtual Ethernet Port */
+	parm.nCapType = GSW_CAP_TYPE_VIRTUAL_PORT;
+	gsw_ops->gsw_common_ops.CapGet(gsw_ops, &parm);
+	cap->info.cap.max_virt_eth_port = parm.nCap;
+
+	/* Total number of DP ports */
+	cap->info.cap.max_num_dp_ports = cap->info.cap.max_eth_port +
+					 cap->info.cap.max_virt_eth_port;
+
+	/* For WLAN 8 bit Mode VAP=11:8/14:8
+	 * For WLAN 9 bit Mode VAP=11:9/14:9
+	 * For Non-WLAN Mode VAP=7:0
+	 */
+	if (cap->info.type == GSWIP32_TYPE) {
+		cap->info.cap.vap_mask[GSW_LOGICAL_PORT_8BIT_WLAN] =
+			VAP_8BIT_MASK_32;
+		cap->info.cap.vap_mask[GSW_LOGICAL_PORT_9BIT_WLAN] =
+			VAP_9BIT_MASK_32;
+	} else {
+		cap->info.cap.vap_mask[GSW_LOGICAL_PORT_8BIT_WLAN] =
+			VAP_8BIT_MASK_31;
+		cap->info.cap.vap_mask[GSW_LOGICAL_PORT_9BIT_WLAN] =
+			VAP_9BIT_MASK_31;
+	}
+	cap->info.cap.vap_offset[GSW_LOGICAL_PORT_8BIT_WLAN] = VAP_8BIT_OFFSET;
+	cap->info.cap.vap_offset[GSW_LOGICAL_PORT_9BIT_WLAN] = VAP_9BIT_OFFSET;
+	cap->info.cap.vap_offset[GSW_LOGICAL_PORT_GPON] = VAP_NONWLAN_OFFSET;
+	cap->info.cap.vap_mask[GSW_LOGICAL_PORT_GPON] = VAP_NONWLAN_MASK;
+
+	cap->info.cap.max_num_learn_limit_port = GSWIP_LEARN_LIMIT_PORT_MAX;
+
+	cap->info.cap.tx_hw_chksum = 0;
+	cap->info.cap.rx_hw_chksum = 0;
+	cap->info.cap.hw_ptp = 1;
+
+	DP_DEBUG(DP_DBG_FLAG_REG, "max_num_subif            = %d\n"
+		 "max_num_bridge_port      = %d\n"
+		 "max_eth_port             = %d\n"
+		 "max_virt_eth_port        = %d\n"
+		 "max_num_dp_ports         = %d\n"
+		 "vap_offset 8-bit Mode    = %d\n"
+		 "vap_mask 8-bit Mode      = %d\n"
+		 "vap_offset 9-bit Mode    = %d\n"
+		 "vap_mask 9-bit Mode      = %d\n"
+		 "vap_offset Non-WLAN mode = %d\n"
+		 "vap_mask Non-WLAN mode   = %d\n",
+		 cap->info.cap.max_num_subif,
+		 cap->info.cap.max_num_bridge_port,
+		 cap->info.cap.max_eth_port,
+		 cap->info.cap.max_virt_eth_port,
+		 cap->info.cap.max_num_dp_ports,
+		 cap->info.cap.vap_offset[GSW_LOGICAL_PORT_8BIT_WLAN],
+		 cap->info.cap.vap_mask[GSW_LOGICAL_PORT_8BIT_WLAN],
+		 cap->info.cap.vap_offset[GSW_LOGICAL_PORT_9BIT_WLAN],
+		 cap->info.cap.vap_mask[GSW_LOGICAL_PORT_9BIT_WLAN],
+		 cap->info.cap.vap_offset[GSW_LOGICAL_PORT_GPON],
+		 cap->info.cap.vap_mask[GSW_LOGICAL_PORT_GPON]);
+
+	return DP_SUCCESS;
+}
+
diff --git a/drivers/net/datapath/dpm/gswip31/datapath_mib.c b/drivers/net/datapath/dpm/gswip31/datapath_mib.c
deleted file mode 100644
index 35523c761cc3..000000000000
--- a/drivers/net/datapath/dpm/gswip31/datapath_mib.c
+++ /dev/null
@@ -1,363 +0,0 @@
-// SPDX-License-Identifier: GPL-2.0
-/* Copyright (C) Intel Corporation
- * Author: Shao Guohua <guohua.shao@intel.com>
- */
-
-#include <net/datapath_api.h>
-#include <net/datapath_proc_api.h>
-#include "../datapath.h"
-
-#define WRAPAROUND32   0xFFFFFFFF
-/*timer interval for mib wraparound handling:
- * Most mib counter is 32 bits, ie, maximu ix 0xFFFFFFFF
- * one pmac port maximum (cpu port) can support less than 3G, ie,
- * 1488096 * 3 packets for 64 bytes case. so the time to have 1 wrapround is:
- * 0xFFFFFFFF / (1488096 * 3) = 962 seconds
- * If each timer check one port and its subif, then 962/16 = 60 sec.
- */
-#define POLL_INTERVAL (60 * HZ)
-#define WAN_EP          15	/*WAN Interface's EP value */
-#define MAX_RMON_ITF    256	/*maximum 256 GSW RMON interface supported */
-
-struct mibs_port {
-	u64 rx_good_bytes;
-	u64 rx_bad_bytes;
-	u64 rx_good_pkts;
-	u64 rx_drop_pkts;
-	/*For eth0_x only, for all others, must keep it
-	 * to zero in order to share same algo
-	 */
-	u64 rx_drop_pkts_pae;
-	u64 rx_disc_pkts_redir;	/*for eth1 only */
-	u64 rx_fcs_err_pkts;
-	u64 rx_undersize_good_pkts;
-	u64 rx_oversize_good_pkts;
-	u64 rx_undersize_err_pkts;
-	u64 rx_oversize_err_pkts;
-	u64 rx_align_err_pkts;
-	u64 rx_filter_pkts;
-
-	u64 tx_good_bytes;
-	u64 tx_good_pkts;
-	u64 tx_drop_pkts;
-	/*For eth0_x only, for all others, must keep it
-	 *to zero in order to share same algo
-	 */
-	u64 tx_drop_pkts_pae;
-	u64 tx_acm_drop_pkts;
-	u64 tx_acm_drop_pkts_pae;	/*for eth0_x only */
-	u64 tx_disc_pkts_redir;	/*for eth1 only */
-	u64 tx_coll_pkts;
-	u64 tx_coll_pkts_pae;	/*for eth0_x only */
-	u64 tx_pkts_redir;	/*for eth1 only */
-	u64 tx_bytes_redir;	/*for eth1 only */
-	u64 rx_fcs_err_pkts_pae;
-	u64 rx_undersize_err_pkts_pae;
-	u64 rx_oversize_err_pkts_pae;
-	u64 rx_align_err_pkts_pae;
-
-	/*driver related */
-	u64 rx_drv_drop_pkts;
-	u64 rx_drv_error_pkts;
-	u64 tx_drv_drop_pkts;
-	u64 tx_drv_error_pkts;
-
-	/*for DSL ATM only */
-	u64 tx_drv_pkts;
-	u64 rx_drv_pkts;
-	u64 tx_drv_bytes;
-	u64 rx_drv_bytes;
-};
-
-struct mib_vap {
-	u64 rx_pkts_itf;
-	u64 rx_disc_pkts_itf;
-	u64 rx_disc_pkts_drv;
-	u64 tx_pkts_itf;
-	u64 tx_disc_pkts_itf;
-	u64 tx_disc_pkts_drv;
-};
-
-struct port_mib {
-	struct mibs_port curr;/*tmp variable used for mib counter calculation */
-	struct mib_vap curr_vap[MAX_SUBIF_PER_PORT];	/*for future */
-};
-
-struct mibs_low_lvl_port {
-	GSW_RMON_Port_cnt_t l;          /*only for ethernet LAN ports */
-	GSW_RMON_Port_cnt_t r;
-	GSW_RMON_Redirect_cnt_t redir; /*only for ethernet WAN port */
-	dp_drv_mib_t drv;
-};
-
-struct mibs_low_lvl_vap {
-	GSW_RMON_If_cnt_t gsw_if; /*for pae only since GSW-L not
-				   *support interface mib
-				   */
-	dp_drv_mib_t drv;
-};
-
-static u32 proc_mib_vap_start_id = 1;
-static u32 proc_mib_vap_end_id = PMAC_MAX_NUM - 1;
-static unsigned long poll_interval = POLL_INTERVAL;
-
-/*save port based lower level last mib counter
- * for wraparound checking
- */
-struct mibs_low_lvl_port last[PMAC_MAX_NUM];
-/*save vap/sub interface based lower level last mib counter
- * for wraparound checking
- */
-struct mibs_low_lvl_vap last_vap[PMAC_MAX_NUM][MAX_SUBIF_PER_PORT];
-/*Save all necessary aggregated basic MIB */
-static struct port_mib aggregate_mib[PMAC_MAX_NUM];
-/*For PAE CPU port only */
-static struct port_mib aggregate_mib_r[1];
-
-#define THREAD_MODE
-
-#ifdef THREAD_MODE
-#include <linux/kthread.h>
-struct task_struct *thread;
-#else
-static struct timer_list exp_timer;	/*timer setting */
-#endif
-
-/*internal API: update local net mib counters periodically */
-static int update_port_mib_lower_lvl(dp_subif_t *subif, u32 flag);
-static int update_vap_mib_lower_lvl(dp_subif_t *subif, u32 flag);
-
-/* ----- API implementation ------- */
-static u64 wraparound(u64 curr, u64 last, u32 size)
-{
-#define WRAPAROUND_MAX_32 0xFFFFFFFF
-	/*for 8 bytes(64bit mib),no need to do wraparound*/
-	if (size > 4 || curr >= last)
-		return curr - last;
-
-	return ((u64)WRAPAROUND_MAX_32) + (u64)curr - last;
-}
-
-static int port_mib_wraparound(u32 ep, struct mibs_low_lvl_port *curr,
-			       struct mibs_low_lvl_port *last)
-{
-	return 0;
-}
-
-static int vap_mib_wraparound(dp_subif_t *subif,
-			      struct mibs_low_lvl_vap *curr,
-			      struct mibs_low_lvl_vap *last)
-{
-	return 0;
-}
-
-static int get_gsw_port_rmon(u32 ep, char *gsw_drv_name,
-			     GSW_RMON_Port_cnt_t *mib)
-{
-	return 0;
-}
-
-static int get_gsw_redirect_rmon(u32 ep, int index,
-				 GSW_RMON_Redirect_cnt_t *mib)
-{
-	return 0;
-}
-
-static int get_gsw_itf_rmon(u32 index, int index,
-			    GSW_RMON_If_cnt_t *mib)
-{
-	return 0;
-}
-
-int get_gsw_interface_base(int port_id)
-{
-	return 0;
-}
-
-/* if ethernet WAN redirect is enabled, return 1,
- * else return 0
- */
-int gsw_eth_wan_redirect_status(void)
-{
-	return 0;
-}
-
-/*Note:
- * Update mib counter for physical port only
- * flag so far no much use only
- */
-static int update_port_mib_lower_lvl(dp_subif_t *subif, u32 flag)
-{
-	return 0;
-}
-
-static void mib_wraparound_timer_poll(unsigned long data)
-{
-}
-
-static int update_vap_mib_lower_lvl(dp_subif_t *subif, u32 flag)
-{
-	return 0;
-}
-
-int dp_reset_sys_mib(u32 flag)
-{
-	return 0;
-}
-
-void proc_mib_timer_read(struct seq_file *s)
-{
-	seq_printf(s, "\nMib timer interval is %u sec\n",
-		   (u32)poll_interval / HZ);
-}
-
-ssize_t proc_mib_timer_write(struct file *file, const char *buf, size_t count,
-			     loff_t *ppos)
-{
-	int len, num;
-	char str[64];
-	char *param_list[2];
-#define MIN_POLL_TIME 2
-	len = (sizeof(str) > count) ? count : sizeof(str) - 1;
-	len -= copy_from_user(str, buf, len);
-	str[len] = 0;
-	num = dp_split_buffer(str, param_list, ARRAY_SIZE(param_list));
-	poll_interval = dp_atoi(param_list[0]);
-
-	if (poll_interval < MIN_POLL_TIME)
-		poll_interval = MIN_POLL_TIME;
-
-	poll_interval *= HZ;
-#ifndef THREAD_MODE
-	mod_timer(&exp_timer, jiffies + poll_interval);
-#endif
-	DP_DEBUG(DP_DBG_FLAG_MIB, "new poll_interval=%u sec\n",
-		 (u32)poll_interval / HZ);
-	return count;
-}
-
-static u32 proc_mib_port_start_id = 1;
-static u32 proc_mib_port_end_id = PMAC_MAX_NUM - 1;
-int proc_mib_inside_dump(struct seq_file *s, int pos)
-{
-	return -1;
-}
-
-int proc_mib_inside_start(void)
-{
-	return proc_mib_port_start_id;
-}
-
-ssize_t proc_mib_inside_write(struct file *file, const char *buf,
-			      size_t count, loff_t *ppos)
-{
-	return count;
-}
-
-/*Note:
- *if (flag & DP_F_STATS_SUBIF), get sub-interface/vap mib only
- *otherwise, get physical port's mib
- */
-int dp_get_port_vap_mib_31(dp_subif_t *subif, void *priv,
-			   struct rtnl_link_stats64 *net_mib, u32 flag)
-{
-	return -1;
-}
-
-/*Clear GSW Interface MIB: only for sub interface/vap only  */
-int clear_gsw_itf_mib(dp_subif_t *subif, u32 flag)
-{
-	return 0;
-}
-
-int dp_clear_netif_mib_31(dp_subif_t *subif, void *priv, u32 flag)
-{
-	return 0;
-}
-
-int proc_mib_port_start(void)
-{
-	return 0;
-}
-
-int proc_mib_port_dump(struct seq_file *s, int pos)
-{
-	return -1;
-}
-
-ssize_t proc_mib_port_write(struct file *file, const char *buf, size_t count,
-			    loff_t *ppos)
-{
-	return count;
-}
-
-int proc_mib_vap_dump(struct seq_file *s, int pos)
-{
-	return -1;
-}
-
-int proc_mib_vap_start(void)
-{
-	return proc_mib_vap_start_id;
-}
-
-ssize_t proc_mib_vap_write(struct file *file, const char *buf, size_t count,
-			   loff_t *ppos)
-{
-	return count;
-}
-
-#ifdef THREAD_MODE
-int mib_wraparound_thread(void *data)
-{
-	while (1) {
-		mib_wraparound_timer_poll(0);
-		msleep(poll_interval / HZ * 1000 / PMAC_MAX_NUM / 2);
-		DP_DEBUG(DP_DBG_FLAG_MIB, "%s\n", __func__);
-	}
-}
-#endif
-
-int adjust_itf(void)
-{
-	return 0;
-}
-
-int set_gsw_itf(u8 ep, u8 ena, int start)
-{
-	return 0;
-}
-
-int reset_gsw_itf(u8 ep)
-{
-}
-
-int dp_mib_init(u32 flag)
-{
-	memset(&aggregate_mib, 0, sizeof(aggregate_mib));
-	memset(&last, 0, sizeof(last));
-	memset(&last_vap, 0, sizeof(last_vap));
-	adjust_itf();
-
-#ifdef THREAD_MODE
-	thread = kthread_run(mib_wraparound_thread, 0, "dp_mib");
-#else
-	init_timer_on_stack(&exp_timer);
-	exp_timer.expires = jiffies + poll_interval;
-	exp_timer.data = 0;
-	exp_timer.function = mib_wraparound_timer_poll;
-	add_timer(&exp_timer);
-	DP_DEBUG(DP_DBG_FLAG_MIB, "%s done\n", __func__);
-#endif
-	return 0;
-}
-
-void dp_mib_exit(void)
-{
-#ifdef THREAD_MODE
-	if (thread)
-		kthread_stop(thread);
-#else
-	del_timer(&exp_timer);
-#endif
-}
diff --git a/drivers/net/datapath/dpm/gswip31/datapath_mib.h b/drivers/net/datapath/dpm/gswip31/datapath_mib.h
deleted file mode 100644
index 40bcd3871e70..000000000000
--- a/drivers/net/datapath/dpm/gswip31/datapath_mib.h
+++ /dev/null
@@ -1,15 +0,0 @@
-// SPDX-License-Identifier: GPL-2.0
-/* Copyright (C) Intel Corporation
- * Author: Shao Guohua <guohua.shao@intel.com>
- */
-
-#ifndef DATAPATH_MIB_H
-#define DATAPATH_MIB_H
-
-int dp_reset_mib(u32 flag);
-int set_gsw_itf(u8 ep, u8 ena, int start);
-int reset_gsw_itf(u8 ep);
-int dp_get_port_vap_mib_31(dp_subif_t *subif, void *priv,
-			   struct rtnl_link_stats64 *net_mib, u32 flag);
-
-#endif
diff --git a/drivers/net/datapath/dpm/gswip31/datapath_misc.c b/drivers/net/datapath/dpm/gswip31/datapath_misc.c
index 0d92ab748a21..2fb5c93ffbda 100644
--- a/drivers/net/datapath/dpm/gswip31/datapath_misc.c
+++ b/drivers/net/datapath/dpm/gswip31/datapath_misc.c
@@ -40,68 +40,6 @@ static struct pon_spl_cfg pon_cfg[] = {
 	},
 };
 
-GSW_QoS_colorRemarkingEntry_t color_remark[] = {
-	{
-		GSW_MARKING_PCP_8P0D,
-		{0, 2, 4, 6, 8, 10, 12, 14, 1, 3, 5, 7, 9, 11, 13, 15},
-	},
-	{
-		GSW_MARKING_PCP_7P1D,
-		{0, 2, 4, 6, 10, 10, 12, 14, 1, 3, 5, 7, 9, 9, 13, 15},
-	},
-	{
-		GSW_MARKING_PCP_6P2D,
-		{0, 2, 6, 6, 10, 10, 12, 14, 1, 3, 5, 5, 9, 9, 13, 15},
-	},
-	{
-		GSW_MARKING_PCP_5P3D,
-		{2, 2, 6, 6, 10, 10, 12, 14, 1, 1, 5, 5, 9, 9, 13, 15},
-	},
-	{
-		GSW_MARKING_DSCP_AF,
-		{10, 18, 26, 34, 34, 34, 34, 34, 12, 20, 28, 36, 36, 36, 36, 36},
-	},
-};
-
-GSW_QoS_colorMarkingEntry_t color_mark[] = {
-	{
-		GSW_MARKING_PCP_5P3D,
-		{0, 0, 0, 0, 2, 2, 2, 2, 4, 4, 4, 4, 6, 6, 7, 7},
-		{Y, Y, G, Y, Y, Y, G, Y, Y, Y, G, Y, G, Y, G, Y},
-	},
-	{
-		GSW_MARKING_PCP_6P2D,
-		{0, 0, 1, 1, 2, 2, 2, 2, 4, 4, 4, 4, 6, 6, 7, 7},
-		{G, Y, G, Y, Y, Y, G, Y, Y, Y, G, Y, G, Y, G, Y},
-	},
-	{
-		GSW_MARKING_PCP_7P1D,
-		{0, 0, 1, 1, 2, 2, 3, 3, 4, 4, 4, 4, 6, 6, 7, 7},
-		{G, Y, G, Y, G, Y, G, Y, Y, Y, G, Y, G, Y, G, Y},
-	},
-	{
-		GSW_MARKING_PCP_8P0D,
-		{0, 0, 1, 1, 2, 2, 3, 3, 4, 4, 5, 5, 6, 6, 7, 7},
-		{G, Y, G, Y, G, Y, G, Y, G, Y, G, Y, G, Y, G, Y},
-	},
-	{
-		GSW_MARKING_DSCP_AF,
-		{
-			0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
-			0, 0, 0, 0, 1, 0, 1, 0, 1, 0, 0, 0, 2, 0,
-			2, 0, 2, 0, 0, 0, 3, 0, 3, 0, 3, 0, 0, 0,
-			0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
-			0, 0, 0, 0, 0, 0, 0, 0
-		},
-		{
-			Y, Y, Y, Y, Y, Y, Y, Y, Y, Y, G, Y, Y, Y, Y, Y,
-			Y, Y, G, Y, Y, Y, Y, Y, Y, Y, G, Y, Y, Y, Y, Y,
-			Y, Y, G, Y, Y, Y, Y, Y, Y, Y, Y, Y, Y, Y, Y, Y,
-			Y, Y, Y, Y, Y, Y, Y, Y, Y, Y, Y, Y, Y, Y, Y, Y
-		},
-	},
-};
-
 static void init_dma_desc_mask(void)
 {
 	struct dma_rx_desc_1 *dw1_rx = (struct dma_rx_desc_1 *)&rx_desc_mask[1];
@@ -172,6 +110,10 @@ void set_desc_pmac_bits(struct dma_tx_desc_0 *dw0,
 #elif IS_ENABLED(CONFIG_INTEL_DATAPATH_HAL_GSWIP32)
 		if (BIT(i) == DESC_REDIR)
 			dw1->field.redir = val;
+		if (BIT(i) == DESC_PMAC)
+			dw1->field.pmac = val;
+		if (BIT(i) == DESC_PREL2)
+			dw1->field.pre_l2 = val;
 #endif
 		if (BIT(i) == DESC_EGP)
 			dw1->field.ep = val;
@@ -236,6 +178,12 @@ static void init_dma_pmac_template(int portid, u32 flags)
 	unsigned long *bits = (unsigned long *)&flags;
 
 	for (i = 0; i < MAX_TEMPLATE; i++) {
+		for (j = 0; j < 4; j++) {
+			memset(&dp_info->desc_dw_templ[j][i], 0, sizeof(u32));
+			memset(&dp_info->desc_dw_mask[j][i], 0, sizeof(u32));
+		}
+		memset(&dp_info->pmac_template[i], 0, sizeof(u32));
+
 		dma_mask_templ(dp_info, i, DESC_DW0_ALL | DESC_DW1_ALL,
 			       0xFFFFFFFF);
 
@@ -960,43 +908,6 @@ int alloc_q_to_port(struct ppv4_q_sch_port *info, u32 flag)
 	return DP_SUCCESS;
 }
 
-static int dp_platform_color_table_set(int inst)
-{
-	struct core_ops *gsw_ops = dp_port_prop[inst].ops[0];
-	GSW_QoS_colorRemarkingEntry_t col_remark = {0};
-	GSW_QoS_colorMarkingEntry_t col_mark = {0};
-	struct qos_ops *gsw_qos;
-	int remark_size = ARRAY_SIZE(color_remark);
-	int mark_size = ARRAY_SIZE(color_mark);
-	int i = 0;
-
-	gsw_qos = &gsw_ops->gsw_qos_ops;
-
-	for (i = 0; i < remark_size; i++) {
-		memset(&col_remark, 0, sizeof(GSW_QoS_colorRemarkingEntry_t));
-		memcpy(&col_remark, &color_remark[i],
-		       sizeof(GSW_QoS_colorRemarkingEntry_t));
-		if (gsw_qos->QOS_ColorReMarkingTableSet(gsw_ops, &col_remark)) {
-			pr_err("%s: GSW_QOS_COLOR_REMARKING_CFG_SET failed\n",
-			       __func__);
-			return DP_FAILURE;
-		}
-	}
-
-	for (i = 0; i < mark_size; i++) {
-		memset(&col_mark, 0, sizeof(GSW_QoS_colorMarkingEntry_t));
-		memcpy(&col_mark, &color_mark[i],
-		       sizeof(GSW_QoS_colorMarkingEntry_t));
-		if (gsw_qos->QOS_ColorMarkingTableSet(gsw_ops, &col_mark)) {
-			pr_err("%s: GSW_QOS_COLOR_REMARKING_CFG_SET failed\n",
-			       __func__);
-			return DP_FAILURE;
-		}
-	}
-
-	return DP_SUCCESS;
-}
-
 #define REINSERT BIT(1)
 
 static
@@ -1032,10 +943,6 @@ int alloc_cpu_q(int inst, struct cbm_cpu_port_data *cpu_data,
 		q_port->tx_ring_size = re_insertion->tx_ring_size;
 
 	} else {
-		if (tx_push->deq_port < 0 ||
-		    tx_push->deq_port >= cap->max_num_deq_ports)
-			return DP_FAILURE;
-
 		q_port->cqe_deq = tx_push->deq_port;
 		q_port->tx_pkt_credit = tx_push->tx_pkt_credit;
 		q_port->tx_ring_addr = (void *)tx_push->txpush_addr_qos;
@@ -1405,7 +1312,8 @@ int dp_platform_queue_set(int inst, u32 flag)
 		}
 	}
 
-	/* Re-Insertion use a New Qid -> Deq Port 23 -> Dma1Tx Ch 16 */
+	/* LGM: Re-Insertion use a New Qid -> Deq Port 28 -> Dma2Tx Ch 1 */
+	/* PRX: Re-Insertion use a New Qid -> Deq Port 23 -> Dma1Tx Ch 16 */
 	if (dp_reinsert_q_map(inst, cpu_port, &cpu_data)) {
 		pr_err("%s: DPM Fail to enable re-insertion Q\n", __func__);
 		return DP_FAILURE;
@@ -1502,7 +1410,7 @@ static int dp_platform_set(int inst, u32 flag)
 			return DP_FAILURE;
 		}
 
-		if (dp_platform_color_table_set(inst)) {
+		if (dp_gsw_color_table_set(inst)) {
 			pr_err("%s: dp_platform_color_table_set Failed\n",
 			       __func__);
 			return DP_FAILURE;
@@ -2653,7 +2561,7 @@ static int dp_ctp_tc_map_set(struct dp_tc_cfg *tc, int flag,
 	return DP_SUCCESS;
 }
 
-int dp_get_reinsert_cnt_31(int inst, int dp_port, int vap, int flag,
+int dp_get_reinsert_cnt(int inst, int dp_port, int vap, int flag,
 			   struct dp_reinsert_count *dp_reins_count)
 {
 	struct pmac_port_info *port_info = get_dp_port_info(inst, dp_port);
@@ -2742,102 +2650,6 @@ static int get_itf_start_end(struct gsw_itf *itf_info, u16 *start, u16 *end)
 	return 0;
 }
 
-static int dp_get_gswip_cap(struct dp_hw_cap *cap, int flag)
-{
-	struct core_ops *gsw_ops;
-	GSW_cap_t parm;
-	GSW_register_t reg;
-
-	gsw_ops = gsw_get_swcore_ops(0);
-	if (!gsw_ops) {
-		pr_err("%s swcore ops is NULL\n", __func__);
-		return DP_FAILURE;
-	}
-
-	reg.nRegAddr = GSWIP_VER_REG_OFFSET;
-	if (gsw_ops->gsw_common_ops.RegisterGet(gsw_ops, &reg))
-		return DP_FAILURE;
-
-	if (reg.nData == 0x031) {
-		cap->info.type = GSWIP31_TYPE;
-		cap->info.ver = GSWIP31_VER;
-	} else if (reg.nData == 0x032) {
-		cap->info.type = GSWIP32_TYPE;
-		cap->info.ver = GSWIP32_VER;
-	}
-
-	/* Max number of CTP */
-	parm.nCapType = GSW_CAP_TYPE_CTP;
-	gsw_ops->gsw_common_ops.CapGet(gsw_ops, &parm);
-	cap->info.cap.max_num_subif = parm.nCap;
-
-	/* Max number of Bridge Port */
-	parm.nCapType = GSW_CAP_TYPE_BRIDGE_PORT;
-	gsw_ops->gsw_common_ops.CapGet(gsw_ops, &parm);
-	cap->info.cap.max_num_bridge_port = parm.nCap;
-
-	/* Max number of Physical Ethernet Port */
-	parm.nCapType = GSW_CAP_TYPE_PORT;
-	gsw_ops->gsw_common_ops.CapGet(gsw_ops, &parm);
-	cap->info.cap.max_eth_port = parm.nCap;
-
-	/* Max number of Virtual Ethernet Port */
-	parm.nCapType = GSW_CAP_TYPE_VIRTUAL_PORT;
-	gsw_ops->gsw_common_ops.CapGet(gsw_ops, &parm);
-	cap->info.cap.max_virt_eth_port = parm.nCap;
-
-	/* Total number of DP ports */
-	cap->info.cap.max_num_dp_ports = cap->info.cap.max_eth_port +
-					 cap->info.cap.max_virt_eth_port;
-
-	/* For WLAN 8 bit Mode VAP=11:8/14:8
-	 * For WLAN 9 bit Mode VAP=11:9/14:9
-	 * For Non-WLAN Mode VAP=7:0
-	 */
-	if (cap->info.type == GSWIP32_TYPE) {
-		cap->info.cap.vap_mask[GSW_LOGICAL_PORT_8BIT_WLAN] = VAP_8BIT_MASK_32;
-		cap->info.cap.vap_mask[GSW_LOGICAL_PORT_9BIT_WLAN] = VAP_9BIT_MASK_32;
-	} else {
-		cap->info.cap.vap_mask[GSW_LOGICAL_PORT_8BIT_WLAN] = VAP_8BIT_MASK_31;
-		cap->info.cap.vap_mask[GSW_LOGICAL_PORT_9BIT_WLAN] = VAP_9BIT_MASK_31;
-	}
-	cap->info.cap.vap_offset[GSW_LOGICAL_PORT_8BIT_WLAN] = VAP_8BIT_OFFSET;
-	cap->info.cap.vap_offset[GSW_LOGICAL_PORT_9BIT_WLAN] = VAP_9BIT_OFFSET;
-	cap->info.cap.vap_offset[GSW_LOGICAL_PORT_GPON] = VAP_NONWLAN_OFFSET;
-	cap->info.cap.vap_mask[GSW_LOGICAL_PORT_GPON] = VAP_NONWLAN_MASK;
-
-	cap->info.cap.max_num_learn_limit_port = GSWIP_LEARN_LIMIT_PORT_MAX;
-
-	cap->info.cap.tx_hw_chksum = 0;
-	cap->info.cap.rx_hw_chksum = 0;
-	cap->info.cap.hw_ptp = 1;
-
-	DP_DEBUG(DP_DBG_FLAG_REG, "max_num_subif            = %d\n"
-		 "max_num_bridge_port      = %d\n"
-		 "max_eth_port             = %d\n"
-		 "max_virt_eth_port        = %d\n"
-		 "max_num_dp_ports         = %d\n"
-		 "vap_offset 8-bit Mode    = %d\n"
-		 "vap_mask 8-bit Mode      = %d\n"
-		 "vap_offset 9-bit Mode    = %d\n"
-		 "vap_mask 9-bit Mode      = %d\n"
-		 "vap_offset Non-WLAN mode = %d\n"
-		 "vap_mask Non-WLAN mode   = %d\n",
-		 cap->info.cap.max_num_subif,
-		 cap->info.cap.max_num_bridge_port,
-		 cap->info.cap.max_eth_port,
-		 cap->info.cap.max_virt_eth_port,
-		 cap->info.cap.max_num_dp_ports,
-		 cap->info.cap.vap_offset[GSW_LOGICAL_PORT_8BIT_WLAN],
-		 cap->info.cap.vap_mask[GSW_LOGICAL_PORT_8BIT_WLAN],
-		 cap->info.cap.vap_offset[GSW_LOGICAL_PORT_9BIT_WLAN],
-		 cap->info.cap.vap_mask[GSW_LOGICAL_PORT_9BIT_WLAN],
-		 cap->info.cap.vap_offset[GSW_LOGICAL_PORT_GPON],
-		 cap->info.cap.vap_mask[GSW_LOGICAL_PORT_GPON]);
-
-	return DP_SUCCESS;
-}
-
 int register_dp_capability(int flag)
 {
 	struct dp_hw_cap cap;
@@ -2905,6 +2717,7 @@ int register_dp_capability(int flag)
 	cap.info.cap.max_num_qos_ports = MAX_PPV4_PORT;
 	cap.info.cap.max_num_subif_per_port = MAX_SUBIF_PER_PORT;
 	cap.info.cap.max_cpu_ports = CQM_MAX_CPU;
+	cap.info.dp_tc_vlan_set = tc_vlan_set;
 
 #if IS_ENABLED(CONFIG_INTEL_DATAPATH_HAL_GSWIP32)
 	cap.info.cap.max_num_deq_ports = DP_MAX_CQM_DEQ;
@@ -2915,7 +2728,6 @@ int register_dp_capability(int flag)
 	cap.info.dp_spl_conn_get = dp_spl_conn_get_32;
 	cap.info.dp_set_io_port = dp_set_io_port;
 	cap.info.dp_get_queue_qocc = dp_get_queue_qocc_32;
-	cap.info.dp_tc_vlan_set = tc_vlan_set_32;
 	cap.info.proc_print_ctp_bp_info = proc_print_ctp_bp_info_32;
 #elif IS_ENABLED(CONFIG_INTEL_DATAPATH_HAL_GSWIP31)
 	cap.info.cap.max_num_deq_ports = MAX_CQM_DEQ;
@@ -2931,7 +2743,6 @@ int register_dp_capability(int flag)
 	cap.info.dp_get_queue_qocc = dp_get_queue_qocc_31;
 	cap.info.swdev_port_learning_limit_set =
 		dp_swdev_port_learning_limit_set;
-	cap.info.dp_tc_vlan_set = tc_vlan_set_31;
 	cap.info.proc_print_ctp_bp_info = proc_print_ctp_bp_info;
 #endif
 
diff --git a/drivers/net/datapath/dpm/gswip31/datapath_misc.h b/drivers/net/datapath/dpm/gswip31/datapath_misc.h
index 9060bd02b159..b194ab17e2be 100644
--- a/drivers/net/datapath/dpm/gswip31/datapath_misc.h
+++ b/drivers/net/datapath/dpm/gswip31/datapath_misc.h
@@ -16,7 +16,6 @@
 #define PAMC_LAN_MAX_NUM	7
 #define VAP_DSL_OFFSET		3
 #define PMAPPER_DISC_CTP	255
-#define MAX_PPV4_PORTS		38
 
 /* For WLAN 8 bit Mode VAP=11:8/14:8
  * For WLAN 9 bit Mode VAP=11:9/14:9
@@ -72,9 +71,6 @@ enum {
 	DESC_PMAC_FLAG_END = BIT(19),
 };
 
-#define Y GSW_DROP_PRECEDENCE_YELLOW
-#define G GSW_DROP_PRECEDENCE_GREEN
-
 /* CQM QMAP mask of all DONTCARE flags */
 #define CQM_QMAP_F_ALL_DONTCARE \
 	(CBM_QUEUE_MAP_F_FLOWID_L_DONTCARE | \
@@ -432,11 +428,11 @@ int get_q_mib(int inst, int qid,
 int get_p_mib(int inst, int pid,
 	      u32 *green /* bytes*/,
 	      u32 *yellow /*bytes*/);
-int tc_vlan_set_31(struct core_ops *ops, struct dp_tc_vlan *vlan,
+int tc_vlan_set(struct core_ops *ops, struct dp_tc_vlan *vlan,
 		   struct dp_tc_vlan_info *info,
 		   int flag);
-int dp_get_reinsert_cnt_31(int inst, int dp_port, int vap, int flag,
-			   struct dp_reinsert_count *dp_reins_count);
+int dp_get_reinsert_cnt(int inst, int dp_port, int vap, int flag,
+			struct dp_reinsert_count *dp_reins_count);
 int dp_queue_block_flush_31(struct dp_qos_blk_flush_queue *cfg,
 			    int flag);
 int dp_port_block_flush_31(struct dp_qos_blk_flush_port *cfg,
@@ -476,4 +472,7 @@ int dp_gsw_del_meter(struct net_device *dev, struct dp_meter_cfg *meter,
 int dp_gsw_reset_mib(int dev, u32 flag);
 int dp_gsw_set_port_lu_md(int inst, u8 ep, u32 flags);
 int dp_gsw_dis_cpu_vlan_md(int inst);
-#endif /* DATAPATH_MISC31_H */
+int dp_get_gswip_cap(struct dp_hw_cap *cap, int flag);
+int dp_gsw_color_table_set(int inst);
+
+#endif
diff --git a/drivers/net/datapath/dpm/gswip31/datapath_tc_asym_vlan.c b/drivers/net/datapath/dpm/gswip31/datapath_tc_asym_vlan.c
index 07179fe14da2..f2526e4cc305 100644
--- a/drivers/net/datapath/dpm/gswip31/datapath_tc_asym_vlan.c
+++ b/drivers/net/datapath/dpm/gswip31/datapath_tc_asym_vlan.c
@@ -1,6 +1,7 @@
 // SPDX-License-Identifier: GPL-2.0
-/* Copyright (C) Intel Corporation
- * Author: Shao Guohua <guohua.shao@intel.com>
+/*
+ * Copyright (C) 2016-2020 Intel Corporation
+ * Copyright (C) 2021 MaxLinear
  */
 
 #include <net/switch_api/gsw_flow_ops.h>
@@ -251,7 +252,7 @@ static int vlan_filter_mode(struct dp_pattern_vlan *pout)
 
 #if IS_ENABLED(CONFIG_PON_QOS)
 static int tc_vlan_filter_untagged_parse(struct dp_tc_vlan *vlan,
-					 int *untagged)
+		int *untagged)
 {
 	struct dp_vlan0 *p0;
 
@@ -285,10 +286,10 @@ static int tc_vlan_filter_untagged_parse(struct dp_tc_vlan *vlan,
 }
 
 static int tc_vlan_filter_single_tagged_parse(struct dp_tc_vlan *vlan,
-					      int *tagged,
-					      GSW_VLANFILTER_config_t *pcfg,
-					      int *j,
-					      int total)
+		int *tagged,
+		GSW_VLANFILTER_config_t *pcfg,
+		int *j,
+		int total)
 {
 	struct dp_vlan1 *p1;
 	int mode = -1;
@@ -357,10 +358,10 @@ static int tc_vlan_filter_single_tagged_parse(struct dp_tc_vlan *vlan,
 }
 
 static int tc_vlan_filter_double_tagged_parse(struct dp_tc_vlan *vlan,
-					      int *tagged,
-					      GSW_VLANFILTER_config_t *pcfg,
-					      int *j,
-					      int total)
+		int *tagged,
+		GSW_VLANFILTER_config_t *pcfg,
+		int *j,
+		int total)
 {
 	struct dp_vlan2 *p2;
 	int mode = -1;
@@ -1080,7 +1081,6 @@ static int tc_ext_vlan(struct core_ops *ops,
 					goto ERROR;
 			}
 		}
-	}
 #else /* IS_ENABLED(CONFIG_PON_QOS) */
 		/* untagged rule */
 		for (i = 0; i < vlan->n_vlan0; i++) {
@@ -1132,8 +1132,8 @@ static int tc_ext_vlan(struct core_ops *ops,
 			if (ret != 0)
 				goto ERROR;
 		}
-	}
 #endif /* IS_ENABLED(CONFIG_PON_QOS) */
+	}
 
 	if ((info->dev_type & 0x01) == 0) {
 		/* Configure CTP */
@@ -1159,7 +1159,7 @@ static int tc_ext_vlan(struct core_ops *ops,
 	return ret;
 }
 
-/*tc_vlan_set_31:
+/*tc_vlan_set:
  *param[in] vlan: directly from DP core API's parameter
  *param[in] info: traslated via vlan->dev.
  *       if it is BP/pmapper dev, info->dev_type=DP_DEV_TYPE_BP_PMAPPER {
@@ -1171,10 +1171,10 @@ static int tc_ext_vlan(struct core_ops *ops,
  *       }
  *flag: reserved for future
  */
-int tc_vlan_set_31(struct core_ops *ops,
-		   struct dp_tc_vlan *vlan,
-		   struct dp_tc_vlan_info *info,
-		   int flag)
+int tc_vlan_set(struct core_ops *ops,
+		struct dp_tc_vlan *vlan,
+		struct dp_tc_vlan_info *info,
+		int flag)
 {
 	/* If it's bridge port, try to configure VLAN filter. */
 	if ((info->dev_type & 0x01) != 0) {
diff --git a/drivers/net/datapath/mib_counters/dp_mib_counters.c b/drivers/net/datapath/mib_counters/dp_mib_counters.c
index 2df57953c920..5ba552e6b42e 100644
--- a/drivers/net/datapath/mib_counters/dp_mib_counters.c
+++ b/drivers/net/datapath/mib_counters/dp_mib_counters.c
@@ -729,7 +729,7 @@ void mib_cnt_get_ethtool_stats_31(struct net_device *dev,
 	read_unlock(&dev_list_lock);
 
 	/* Get reinserted packets counters */
-	(void)dp_get_reinsert_cnt_31(mib_cnt->inst,
+	(void)dp_get_reinsert_cnt(mib_cnt->inst,
 				     mib_cnt->port_id,
 				     mib_cnt->sub_if_id_grp,
 				     0,
